java/email-contact-form/src/Main.java

```diff
@@ -0,0 +1,133 @@
+import io.openruntimes.java.RuntimeOutput;
+import java.util.Map;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.lang.*;
+import io.appwrite.Client;
+import jakarta.mail.MessagingException;
+
+
+public class Main {
+
+    public RuntimeOutput main(RuntimeContext context) throws Exception {
+
+        List<String> requiredEnvVariables = Arrays.asList(
+                "SUBMIT_EMAIL",
+                "SMTP_HOST",
+                "SMTP_USERNAME",
+                "SMTP_PASSWORD"
+        );
+
+        Utils.throwIfMissing(System.getenv(), requiredEnvVariables);
+        if (System.getenv("ALLOWED_ORIGINS").equals("*")) {
+            context.log("WARNING: Allowing requests from any origin - this is a security risk!");
+        }
+
+        if (context.getReq().getMethod().equals("GET")) {
+            return context.getRes().send(
+                    Utils.getHtmlContent("index.html"),
+                    200,
+                    Map.of("content-type", "text/html")
+            );
+        }
+
+
+        if(!context.getReq().getHeaders().get("content-type").equals("application/x-www-form-urlencoded")){
+            context.error("Incorrect content type");
+            String referer = context.getReq().getHeaders().get("referer");
+            String errorCode = ErrorCode.INVALID_REQUEST;
+            return context.getRes().redirect(
+                    String.format("%s?code=%s", referer, errorCode)
+            );
+        }
+
+        if(!Cors.isOriginPermitted(context)){
+            context.error("Origin not permitted");
+            String referer = context.getReq().getHeaders().get("referer");
+            String errorCode = ErrorCode.INVALID_REQUEST;
+            return context.getRes().redirect(
+                    String.format("%s?code=%s", referer, errorCode)
+            );
+        }
+
+        Map<String, List<String>> formData = new HashMap<>();
+        String body = (String) context.getReq().getBody();
+        String[] params = body.split("&");
+        for (String param : params) {
+            String[] keyValue = param.split("=");
+            String key = keyValue[0];
+            String value = keyValue[1];
+            if (!formData.containsKey(key)) {
+                formData.put(key, new ArrayList<>());
+            }
+            formData.get(key).add(value);
+        }
+
+        Map<String, String> form = new HashMap<>();
+        for (Map.Entry<String, List<String>> entry : formData.entrySet()) {
+            form.put(entry.getKey(), entry.getValue().get(0));
+        }
+
+        try {
+            Utils.throwIfMissing(form, Collections.singletonList("email"));
+        } catch (IllegalArgumentException ex) {
+            String referer = context.getReq().getHeaders().get("referer");
+            String errorCode = ErrorCode.MISSING_FORM_FIELDS;
```

Comment:
I don't think we need this variable, just move it inline

Reply:
Removed the errorCode variable and moved errorCode it to inline